#! /bin/sh /usr/share/dpatch/dpatch-run
## 01_allow_dark_themes.dpatch by  <albertomilone@alice.it>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Allow nvidia-settings to work with dark backgrounds

@DPATCH@
diff -urNad nvidia-settings-173.14.09~/src/gtk+-2.x/ctkglx.c nvidia-settings-173.14.09/src/gtk+-2.x/ctkglx.c
--- nvidia-settings-173.14.09~/src/gtk+-2.x/ctkglx.c	2008-06-05 02:14:03.000000000 -0500
+++ nvidia-settings-173.14.09/src/gtk+-2.x/ctkglx.c	2008-08-28 16:59:43.000000000 -0500
@@ -265,8 +265,8 @@
     event = gtk_event_box_new();
     gtk_scrolled_window_set_policy(GTK_SCROLLED_WINDOW(scrollWin),
                                    GTK_POLICY_NEVER, GTK_POLICY_ALWAYS);
-    gtk_widget_modify_fg(event, GTK_STATE_NORMAL, &(event->style->text[GTK_STATE_NORMAL]));
-    gtk_widget_modify_bg(event, GTK_STATE_NORMAL, &(event->style->base[GTK_STATE_NORMAL]));
+    //gtk_widget_modify_fg(event, GTK_STATE_NORMAL, &(event->style->text[GTK_STATE_NORMAL]));
+    //gtk_widget_modify_bg(event, GTK_STATE_NORMAL, &(event->style->base[GTK_STATE_NORMAL]));
     gtk_container_add(GTK_CONTAINER(event), hbox);
     gtk_scrolled_window_add_with_viewport(GTK_SCROLLED_WINDOW(scrollWin),
                                           event);
@@ -349,15 +349,15 @@
     data_table = gtk_table_new(num_fbconfigs, NUM_FBCONFIG_ATTRIBS, FALSE);
     event = gtk_event_box_new();
     
-    gtk_widget_modify_fg(data_table, GTK_STATE_NORMAL, 
-                         &(data_table->style->text[GTK_STATE_NORMAL]));
-    gtk_widget_modify_bg(data_table, GTK_STATE_NORMAL, 
-                         &(data_table->style->base[GTK_STATE_NORMAL]));
+    //gtk_widget_modify_fg(data_table, GTK_STATE_NORMAL, 
+    //                     &(data_table->style->text[GTK_STATE_NORMAL]));
+    //gtk_widget_modify_bg(data_table, GTK_STATE_NORMAL, 
+    //                     &(data_table->style->base[GTK_STATE_NORMAL]));
     gtk_container_add (GTK_CONTAINER(event), data_table);
-    gtk_widget_modify_fg(event, GTK_STATE_NORMAL, 
-                         &(event->style->text[GTK_STATE_NORMAL]));
-    gtk_widget_modify_bg(event, GTK_STATE_NORMAL, 
-                         &(event->style->base[GTK_STATE_NORMAL]));
+    //gtk_widget_modify_fg(event, GTK_STATE_NORMAL, 
+    //                     &(event->style->text[GTK_STATE_NORMAL]));
+    //gtk_widget_modify_bg(event, GTK_STATE_NORMAL, 
+    //                     &(event->style->base[GTK_STATE_NORMAL]));
     gtk_container_add(GTK_CONTAINER(data_viewport), event);
 
     /* Pack the fbconfig header and data tables */
